src/@types/matrix-js-sdk.d.ts

```diff
@@ -50,4 +51,35 @@ declare module "matrix-js-sdk/src/types" {
         };
         "m.room.bot.options": unknown;
     }
+
+    export interface TimelineEvents {
+        "io.element.performance_metric": {
+            "io.element.performance_metrics": {
+                forEventId: string;
+                responseTs: number;
+                kind: "send_time";
+            };
+        };
+    }
+
+    export interface AudioContent {
+        // MSC1767 + Ideals of MSC2516 as MSC3245
+        // https://github.com/matrix-org/matrix-doc/pull/3245
+        "org.matrix.msc1767.text"?: string;
+        "org.matrix.msc1767.file"?: {
+            url?: string;
+            file?: EncryptedFile;
+            name: string;
+            mimetype: string;
+            size: number;
+        };
+        "org.matrix.msc1767.audio"?: {
+            duration: number;
+            // https://github.com/matrix-org/matrix-doc/pull/3246
+            waveform?: number[];
+        };
+        "org.matrix.msc3245.voice"?: {};
+
+        "io.element.voice_broadcast_chunk"?: { sequence: number };
+    }
```

Comment:
I'm not loving all this. It feels like a bit of a dumping ground. What does it all mean? Why is it not documented? Why, if we need it, isn't it in the js-sdk itself?

Reply:
Because the js-sdk doesn't know anything about `io.element.*` events and whatever other MSC events and similar extensions apps choose to built atop of it. It is intended to be a generic library to be shared by a plethora of projects. These interfaces leverage Typescript declaration merging to be able to extend the js-sdk's types without needing casts all over the shop.

Reply:
Right yes, I can see this doesn't belong in js-sdk.

> These interfaces leverage Typescript declaration merging to be able to extend the js-sdk's types without needing casts all over the shop.

Ok well, I guess it would be nice if there was a comment explaining that that is what they are doing.

Reply:
Done